ray-operator/controllers/ray/rayjob_controller_unit_test.go

```diff
@@ -624,3 +625,125 @@ func TestEmitRayJobExecutionDuration(t *testing.T) {
 		})
 	}
 }
+
+func TestReconcile_ValidationFailures(t *testing.T) {
+	scheme := runtime.NewScheme()
+	_ = rayv1.AddToScheme(scheme)
+	_ = corev1.AddToScheme(scheme)
+
+	tests := []struct {
+		name          string
+		rayJob        *rayv1.RayJob
+		shouldContain string
+	}{
+		{
+			name: "metadata validation error - name too long",
+			rayJob: &rayv1.RayJob{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      strings.Repeat("a", 48), // Exceeds max length of 47
+					Namespace: "default",
+				},
+				Spec: rayv1.RayJobSpec{
+					RayClusterSpec: &rayv1.RayClusterSpec{
+						HeadGroupSpec: rayv1.HeadGroupSpec{
+							Template: corev1.PodTemplateSpec{
+								Spec: corev1.PodSpec{
+									Containers: []corev1.Container{{Image: "rayproject/ray"}},
+								},
+							},
+						},
+					},
+				},
+			},
+			shouldContain: "The RayJob metadata is invalid",
```

Comment:
Would it be difficult to include the message of error in `shouldContain`? That could be more specific that it did cause a certain error.